src/tiered.js

```diff
@@ -23,80 +21,50 @@ class TieredDatastore /* :: <Value> */ {
     this.stores = stores.slice()
   }
 
-  open (callback /* : Callback<void> */) /* : void */ {
-    each(this.stores, (store, cb) => {
-      store.open(cb)
-    }, (err) => {
-      if (err) {
-        return callback(Errors.dbOpenFailedError())
-      }
-      callback()
-    })
+  async open () /* : void */ {
+    try {
+      await (this.stores.map((store) => store.open()))
+    } catch (err) {
```

Comment:
I know this wasn't part of the original code, but I wonder if we should log the error here?